pkg/generators/markers.go

```diff
@@ -305,20 +336,25 @@ func putNestedValue(m map[string]any, k []string, v any) (map[string]any, error)
 		var arrayDestination []any
 		if existing, ok := m[key]; !ok {
 			arrayDestination = make([]any, index+1)
-		} else {
+		} else if existing, ok := existing.([]any); !ok {
+			// Error case. Existing isn't of correct type. Can happen if
+			// someone is subscripting a field that was previously not an array
+			return nil, fmt.Errorf("expected []any at key %v, got %T", key, existing)
+		} else if index >= len(existing) {
 			// Ensure array is big enough
-			arrayDestination = append(existing.([]any), make([]any, index-len(existing.([]any))+1)...)
+			arrayDestination = append(existing, make([]any, index-len(existing)+1)...)
+		} else {
+			arrayDestination = existing
 		}
 
 		m[key] = arrayDestination
 		if arrayDestination[index] == nil {
 			// Doesn't exist case
 			destination := make(map[string]any)
-			arrayDestination[index] = destination
-			return putNestedValue(destination, rest, v)
+			arrayDestination[index], err = putNestedValue(destination, rest, v)
 		} else if dst, ok := arrayDestination[index].(map[string]any); ok {
 			// Already exists case, correct type
-			return putNestedValue(dst, rest, v)
+			arrayDestination[index], err = putNestedValue(dst, rest, v)
 		}
 
 		// Already exists, incorrect type. Error
```

Comment:
This comment is inaccurate now that we're not returning on line 354 and 357? I'm also a bit confused why we're ignoring the error returned from those lines now?

Reply:
nice catch. I was throwing out the error for the recursie call. Ive straightened out the error handling a bit.

Reply:
awesome, thanks!